Linear Arc Shots implemented

BMStrom opened the pull request:
Linear Arc Shots effect (an effect that basically applies the Linear Arc modifier to the player's projectiles until it wears off) created/implemented

BMStrom committed:
Linear Arc Shots implemented

Linear Arc Shots effect created and implemented

Miczu commented:
To make the auto update work, the new files need to be added to the "manifest.json".
To detect changes, the files use sha256 to compare current file to the one defined in the manifest.
All new files need to be included, or mod user won't have all files needed to run the mod.

There are multiple ways to generate sha256 for a file, but one quick way is to download a file from github (in case there are some end of line changes made by github etc.) and use one of the online tools like this one https://www.strerr.com/en/sha256_file.html

Miczu commented:
You don't need to close PR to add additional files/fixes.

BMStrom committed:
Changed manifest for Linear Arc Shots

Changed manifest file for Linear Arc Shots effect

BMStrom committed:
Updated manifest file

Updated manifest file for Linear Arc Shots effect

BMStrom committed:
removed unnecessary file

removed unnecessary file that was created accidentally

Miczu commented:
Looks like one of the files has a hash mismatch
![image](https://github.com/user-attachments/assets/7cc1d2a4-317e-4301-8f20-0c0397e7b8e5)
Either it was hash from different file or your local file isn't the same as the one returned by github (feel free to download raw file from github and use that one).

BMStrom committed:
Corrected manifest file

Corrected manifest file with correct hash

Miczu commented:
Sadly the start_twitch.bat only shows one manifest error at a time and stops :(
Here is the next one:
![image](https://github.com/user-attachments/assets/92f31c5e-998b-4e68-b4ef-46dd1cde4c44)

BMStrom committed:
Update manifest.json

Miczu commented:
It feels like giving player 8 valid directions to shoot is too generous / low impact.

Let's break down casting spells into 3 categories:
- killing enemies - bypassing enemies and running past is already superior strategy, so linear arc doesn't have that big of an impact
- digging - no effect on dropping bombs, digger bolts. Rockets could be affected, but 8 directions don't force player into too hard of a position. 4 directions would force player to be directly below / to the side.
- movement - teleport bolts moving horizontally, diagonally and up/down is basically full movement. Limiting spell directions into 4 directions would force more of the teleport aiming to be compensated by hovering - probably good nerf to tp and interesting to see how player compensate for the limitation.

Do you mind updating the code to limit it into 4 directions and updating manifest afterwards? 🙏

BMStrom committed:
Reduced Line Arc Shots directions

Changed so that Line Arc Shots only allow you to shoot in cardinal directions

BMStrom committed:
Updated manifest

Updated manifest to account for changes to Linear Arc Shots effect

Miczu commented:
I'll test after work.
Do you want to update your descriptions?
```--Linear Arc Shots
--Makes projectiles fire only in cardinal and diagonal directions
--curses
--100
--Your projectiles only fire in cardinal and diagonal directions```

BMStrom commented:
> I'll test after work. Do you want to update your descriptions?
> 
> ```
> --Makes projectiles fire only in cardinal and diagonal directions
> --curses
> --100
> --Your projectiles only fire in cardinal and diagonal directions```
> ```

Sure. I forgot to do that. I'll do it now.
By the way, I keep having to change the scripts and update the manifest separately, because when I try to make a hash with my local copies of the files it generates an incorrect hash.

BMStrom committed:
Changed description of Linear Arc Shots effect

Changed description of Linear Arc Shots effect so it properly describes the change to the Linear Arc effect

BMStrom committed:
Updated manifest

Updated manifest

Miczu reviewed (approved):
Looks good to me
